gr-network/lib/udp_sink_impl.cc

```diff
@@ -224,15 +216,17 @@ int udp_sink_impl::work(int noutput_items,
     const char* in = (const char*)input_items[0];
 
     // Build a long local queue to pull from so we can break it up easier
-    for (int i = 0; i < num_bytes_to_transmit; i++) {
-        d_localqueue->push_back(in[i]);
-    }
+    if (d_localqueue_writer->space_available() < num_bytes_to_transmit)
+        d_localqueue_reader->update_read_pointer(num_bytes_to_transmit -
+                                                 d_localqueue_writer->space_available());
+    memcpy(d_localqueue_writer->write_pointer(), in, num_bytes_to_transmit);
```

Comment:
Bounds check on `num_bytes_to_transmit` here? The original code didn't need one because it woudl just keep wrapping.

Reply:
Good catch. I'll throw in bounds checks.